app/validators/your_details_completion_validator.rb

```diff
@@ -1,14 +1,24 @@
 class YourDetailsCompletionValidator < ActiveModel::EachValidator
-  def validate_each(record, attribute, _application_choice)
+  include Rails.application.routes.url_helpers
+
+  def validate_each(record, attribute, application_choice)
     presenter = CandidateInterface::ApplicationFormPresenter.new(record.application_form)
```

Comment:
This is the type of logic that we need to revisit, I had to take a decision to use the `CandidateInterface::ApplicationFormPresenter` to verify all sections are complete but we might need to rewrite this and this validator becomes more difficult to read.
We might document somewhere on places we need to comeback after 10th October.

Reply:
Thinking quickly about design I was thinking in an interface on something like these:

```
  CandidateInterface::ApplicationFormSections.new(application_form:, application_choice:).incomplete_sections
  => [<#CandidateInterface::ApplicationFormSection name='personal_statement'>]

CandidateInterface::ApplicationFormSections.new(application_form:, application_choice:).complete_sections
 => [<#CandidateInterface::ApplicationFormSection name='degrees'>]
```

Reply:
Something for the future though